cmake/process_options.cmake

```diff
@@ -76,6 +78,14 @@ if(ENABLE_NATIVE)
     target_compile_options(lightning_compile_options INTERFACE -march=native)
 endif()
 
+if(ENABLE_MPI)
+    set(ENABLE_OPENMP OFF)
```

Comment:
Where these incompatible?

Reply:
It could be the conflicts between openmp and multithreading. Since we've removed `OPENMP` for the serial backend, let's remove it here.